scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -192,6 +193,8 @@ sealed trait Execution[+T] extends java.io.Serializable {
    */
   def zip[U](that: Execution[U]): Execution[(T, U)] =
     Zipped(this, that)
+
+  override val hashCode = super.hashCode
```

Comment:
yeah, using `lazy val hashCode = ScalaRunTime._hashCode(this)` might work. I'm nervous about putting a `val` there due to init order bugs.

Yes, by `strict` I meant eager/val or even lazy val (which means strict is bad terminology to use). Just don't want recursion is all.